### PR TITLE
HLS support

### PR DESCRIPTION
**Pull request by L-as**

### CODE REVIEW

**Comment by L-as** [comment]
It doesn't work for some reason. The error message indicates that neither the custom source nor the flags have been applied.

**Comment by L-as** [comment]
https://hercules-ci.com/accounts/github/Plutonomicon/derivations/%2Fnix%2Fstore%2F4xf1mhfc52nl5y9zgspzcrjyghpm999b-cabal-configure-error.drv/log?via-job=09034b16-cc8e-4f4e-9323-99818d521093

**Comment by L-as** [comment]
https://github.com/input-output-hk/haskell.nix/issues/1272

**Comment by L-as** [comment]
I finally fixed this. The issue was in haskell.nix: https://github.com/input-output-hk/haskell.nix/pull/1355

**Comment by L-as** [comment]
@srid @Geometer1729 please test

**Comment by L-as** [comment]
CI failure is unrelated to this, the derivation for the shell works fine.

File: flake.nix [hunk 1]
```diff
@@ -131,7 +119,74 @@
...
+        cabalProjectLocal = ''
+          allow-newer: *:*
+
+          constraints:
+            primitive-unlifted < 1.0.0.0
+
+          package haskell-language-server
+            flags: +use-ghc-stub +pedantic +ignore-plugins-ghc-bounds -alternateNumberFormat -brittany -callhierarchy -class -eval -floskell -fourmolu -haddockComments -hlint -importLens -ormolu -refineImports -retrie -splice -stylishhaskell -tactic -importLens
+
+        '';
```
Comment (srid) [inline flake.nix:187]: Where does this `cabal.project` come from?

**Review by srid — APPROVED** [review]
@L-as Thanks! HLS works for me on this PR.

**Comment by L-as** [comment]
I looked at https://github.com/haskell/haskell-language-server/blob/master/stack-9.2.1.yaml to figure out which flags to pass.

**Pull request merged**